lib/internal/modules/cjs/loader.js

```diff
@@ -113,6 +113,10 @@ const policy = getOptionValue('--experimental-policy') ?
   require('internal/process/policy') :
   null;
 
+const permission = getOptionValue('--experimental-permission') ?
+  require('internal/process/permission') :
+  null;
```

Comment:
I think I read somewhere that using `getOptionValue` at the top-level was bad practice for snapshotting. //cc @joyeecheung

Reply:
Actually, that's the reason I created the `isEnabled` for lazy loading. We have a test that validates `getOptionValue` isn't called at the top level.

Reply:
This isn't lazy loading, as getOptionValue() would still be invoked when the CJS is loaded, it doesn't fail now because the CJS loader is not yet in the built-in snapshot (but it should, and will be in https://github.com/nodejs/node/pull/45849). You can get around by moving this statement to anywhere you actually need the permission module. Loading an already loaded module is not really that expensive anyway (compared to all the stuff we do around that).

Reply:
Or actually you can just do `require('internal/process/permission')` unconditionally here since that module isn't querying CLI options when it's evaluated, and I think it's worth being included into the snapshot. Then you can do `if (permission.isEnabled() && !permission.check(..))`  within functions that are actually invoked at run time, which is what lazy means - you don't have to query the CLI arguments the second that we evaluate the CJS loader module (which basically just contains a bunch of functions to be called later at runtime), you only need to query the CLI arguments when you actually need to make a decision in the code.

Reply:
Yes, you are right. I confused this file with the `lib/fs.js` which already implements this approach.